### PR TITLE
Allows job parameters to be logged only as system message or not at all

### PR DESCRIPTION
**Pull request by meg1502**
Altough parameters can be restricted by permission within the collectParameters method, once they are logged they can be seen by everyone who has access to these logs. As this could expose sensitive data, parameters can now restrict their logging behaviour.

Fixes: OX-6144

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

**Review by fhaScireum — APPROVED** [review]

File: src/main/java/sirius/biz/jobs/batch/BatchProcessJobFactory.java [hunk 1]
```diff
@@ -213,16 +216,36 @@ protected void executeTask(ProcessContext process) throws Exception {
...
+
+        if (parameters.stream().anyMatch(Parameter::shouldBeSystemLogged)) {
+            logSystemParameters(process, parameters);
+        }
+    }
 
-        getParameters().forEach(param -> {
+    private void logSystemParameters(ProcessContext process, List<Parameter<?, ?>> parameters) {
+        StringBuilder output = new StringBuilder();
+        output.append("Systemparameter:\n\n");
```
Comment (andyHa) [inline src/main/java/sirius/biz/jobs/batch/BatchProcessJobFactory.java:233]: i18n

File: src/main/java/sirius/biz/jobs/batch/BatchProcessJobFactory.java [hunk 2]
```diff
@@ -213,16 +216,36 @@ protected void executeTask(ProcessContext process) throws Exception {
     protected abstract BatchJob createJob(ProcessContext process) throws Exception;
 
     protected void logParameters(ProcessContext process) {
+        List<Parameter<?, ?>> parameters = getParameters();
+
         StringBuilder output = new StringBuilder();
         output.append("Parameter:\n\n");
```
Comment (andyHa) [inline src/main/java/sirius/biz/jobs/batch/BatchProcessJobFactory.java:222]: i18n

Comment (andyHa) [reply src/main/java/sirius/biz/jobs/batch/BatchProcessJobFactory.java:222]: couldn't we re-use the code and call it twice with a different heading and predicate (`Parameter::shouldBeSystemLogged` or `p -> !p.shouldBeSystemLogged()`)

File: src/main/java/sirius/biz/jobs/params/Parameter.java [hunk 3]
```diff
@@ -122,6 +148,25 @@ public boolean isVisible(Map<String, String> context) {
         return get(context).isPresent();
     }
 
+    /**
```
Comment (andyHa) [inline src/main/java/sirius/biz/jobs/params/Parameter.java:151]: wouldn't a getter be enough here? it is only used in another condition, we could even use this as a parameter (s.a.)

File: src/main/java/sirius/biz/jobs/batch/BatchProcessJobFactory.java (hunk 2 above)
Comment (andyHa) [reply src/main/java/sirius/biz/jobs/batch/BatchProcessJobFactory.java:222]: scratch that - add a getVisibility() +- parameter and simply pass in the desired visibility type (other normal or system)...

**Review by andyHa — CHANGES_REQUESTED** [review]
...

**Review by idlira — APPROVED** [review]

**Review by bwiedmann — APPROVED** [review]

**Review by qw3ry — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Pull request merged**